storage/storage_test.go

```diff
@@ -0,0 +1,56 @@
+package storage
+
+import (
+	"context"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"strings"
+
+	"github.com/evergreen-ci/pail"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func copyDirectory(source string, target string) error {
+	var err error = filepath.Walk(source, func(path string, info fs.FileInfo, err error) error {
+		var relPath string = strings.Replace(path, source, "", 1)
+		if relPath == "" {
+			return nil
+		}
+		if info.IsDir() {
+			return os.Mkdir(filepath.Join(target, relPath), 0755)
+		} else {
+			var data, fileReadErr = ioutil.ReadFile(filepath.Join(source, relPath))
+			if fileReadErr != nil {
+				return fileReadErr
+			}
+			return ioutil.WriteFile(filepath.Join(target, relPath), data, 0666)
+		}
+	})
+	return err
+}
+
+func makeTestStorage(t *testing.T, initDir string) Storage {
+	os.Mkdir("../_bucketdata", 0755)
+	copyDirectory(initDir, "../_bucketdata")
+	bucket, err := pail.NewLocalBucket(pail.LocalOptions{
+		Path:   "../_bucketdata",
```

Comment:
Do we need to copy the data into a new directory? Could the bucket just be pointed at `testdata`?

Reply:
I thought about this. My concern had been if we want to write to the bucket, it'd be easier just to designate a temporary folder for doing test operations. I also considered just using a literal temporary folder for this, if we think that would be better?

Reply:
Actually, I just realized that we _are_ going to need to copy the whole file structure, but we should use pail's synchronization logic to do it, since at some point we're going to need to upload the test files to S3 for testing purposes.

Reply:
This is updated to upload via pail's "push" method.

Reply:
nice!